Flaky tests: KernelCommunicationTests assess read + drops lower bound

See:
```
Running kernel tests requires root.
[==========] Running 1 test from 1 test case.
[----------] Global test environment set-up.
[----------] 1 test from KernelCommunicationTests
[ RUN      ] KernelCommunicationTests.test_communication
/Users/osquery/jenkins/workspace/osqueryPullRequestBuild/TargetSystem/osx11/osquery/events/kernel/tests/kernel_communication_tests.cpp:90: Failure
Value of: reads + drops
  Actual: 1995610
Expected: num_threads * events_per_thread
Which is: 2000000
[  FAILED  ] KernelCommunicationTests.test_communication (6338 ms)
[----------] 1 test from KernelCommunicationTests (6338 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test case ran. (6338 ms total)
[  PASSED  ] 0 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] KernelCommunicationTests.test_communication

 1 FAILED TEST
make[2]: *** [kernel/CMakeFiles/kernel-test] Error 1
make[1]: *** [kernel-test/fast] Error 2
make: *** [kernel-test/fast] Error 2
```

This test failure is caused by the default mode for kernel event consumption switching from lossless (blocking) to lossy.

--- a/osquery/events/kernel/tests/kernel_communication_tests.cpp
+++ b/osquery/events/kernel/tests/kernel_communication_tests.cpp
@@ -87,7 +87,12 @@ TEST_F(KernelCommunicationTests, test_communication) {
     }
   } while (tasks > 0);
 
-  EXPECT_EQ(num_threads * events_per_thread, reads + drops);
+  auto total_events = reads + drops;
+  auto expected_events = num_threads * events_per_thread;
+
+  // Since the sync is opened non-blocking we allow a 5% drop rate.
+  EXPECT_GT(total_events, expected_events * 0.95);
+  EXPECT_LE(total_events, expected_events);
 }
 #endif // KERNEL_TEST
 }

